bevy_kayak_ui/src/lib.rs

```diff
@@ -40,8 +42,15 @@ pub fn update(world: &mut World) {
         if let Ok(mut context) = bevy_context.kayak_context.write() {
             context.set_global_state(std::mem::take(world));
             context.render();
-            *world = context.take_global_state::<World>().unwrap()
+            *world = context.take_global_state::<World>().unwrap();
+
+            if let Some(ref mut windows) = world.get_resource_mut::<Windows>() {
+                if let Some(window) = windows.get_primary_mut() {
+                    window.set_cursor_icon(convert_cursor_icon(context.cursor_icon()));
+                }
+            }
```

Comment:
Alternatively, we could maybe register a callback for when the cursor icon changes. But this was the simplest method.